### PR TITLE
LBSD-1151: Allow consumer to have syndicated posts from a specific date

### PR DESCRIPTION
**Pull request by mscam**
- Added _start_date_ to _syndication_in_ & _syndication_out_ schema.
- Added PATCH support to _blogs_ and _syndication_ blueprints.
- Added behave test for _blogs_ syndicate blueprint.

### CODE REVIEW

**Comment by mscam** [comment]
Reminder: please remember to send the correct date format as specified by python-eve's _DATE_FORMAT_ configuration setting http://python-eve.org/config.html

File: server/liveblog/syndication/blogs.py [hunk 1]
```diff
@@ -107,16 +129,26 @@ def _delete_blogs_syndicate(blog_id, consumer_blog_id):
...
 
-@blogs_blueprint.route('/api/syndication/blogs/<string:blog_id>/syndicate', methods=['POST', 'DELETE'])
+@blogs_blueprint.route('/api/syndication/blogs/<string:blog_id>/syndicate', methods=['POST', 'PATCH', 'DELETE'])
 def blogs_syndicate(blog_id):
-    consumer_blog_id = request.get_json().get('consumer_blog_id')
+    data = request.get_json()
+    start_date = data.get('start_date')
+    if start_date:
+        try:
+            start_date = str_to_date(start_date)
```
Comment (djbrd-sourcefabric) [inline server/liveblog/syndication/blogs.py]: Might be nice to document somewhere, even if just in a comment, what is the expected format

File: server/liveblog/syndication/producer.py [hunk 2]
```diff
@@ -189,6 +198,29 @@ def _create_producer_blogs_syndicate(producer_id, blog_id, consumer_blog_id, aut
...
+    try:
+        response = producers.syndicate(producer_id, blog_id, consumer_blog_id, start_date, update=True,
+                                       json_loads=False)
+    except ProducerAPIError as e:
+        return api_response(str(e), 500)
+    else:
+        if response.status_code == 200:
+            syndication_in = in_service.get_syndication(producer_id, blog_id, consumer_blog_id)
+            in_service.update(syndication_in['_id'], {'start_date': start_date}, syndication_in)
+            del syndication_in['blog_token']
```
Comment (djbrd-sourcefabric) [inline server/liveblog/syndication/producer.py]: This looks a little dangerous. Should be more difficult for the blog_token to be returned to the client. Do we actually need it to be exposed at all?

File: server/liveblog/syndication/blogs.py [hunk 3]
```diff
@@ -76,20 +78,40 @@ def _create_blogs_syndicate(blog_id, consumer_blog_id):
...
         'blog_id': blog_id,
         'consumer_id': consumer_id,
-        'consumer_blog_id': consumer_blog_id
-    }])[0]
+        'consumer_blog_id': consumer_blog_id,
+        'start_date': start_date
+    }
+
+    syndication_id = out_service.post([doc])[0]
     syndication = out_service.find_one(_id=syndication_id, req=None)
```
Comment (mscam) [inline server/liveblog/syndication/blogs.py]: unfortunately it's required as post() method is returning the document id, not the whole document, and we need the whole document to get and put the generated token in the response.

File: server/liveblog/syndication/producer.py [hunk 4]
```diff
@@ -213,17 +245,28 @@ def _delete_producer_blogs_syndicate(producer_id, blog_id, consumer_blog_id):
...
+    if start_date:
+        try:
+            start_date = str_to_date(start_date)
+        except ValueError:
+            return api_error('start_date is not valid.', 400)
+
     if request.method == 'DELETE':
         return _delete_producer_blogs_syndicate(producer_id, blog_id, consumer_blog_id)
+    elif request.method == 'PATCH':
+        return _update_producer_blogs_syndicate(producer_id, blog_id, consumer_blog_id, start_date)
```
Comment (djbrd-sourcefabric) [inline server/liveblog/syndication/producer.py]: How is auto_publish value updated?

File: server/liveblog/syndication/syndication.py [hunk 5]
```diff
@@ -33,11 +33,13 @@
...
     },
-    'last_delivered_post_id': Resource.rel('posts', embeddable=True, required=True,
-                                           nullable=True, type="objectid"),
     'token': {
         'type': 'string',
         'unique': True
+    },
+    'start_date': {
+        'type': 'datetime',
+        'default': None
```
Comment (djbrd-sourcefabric) [inline server/liveblog/syndication/syndication.py]: Default to now?

File: server/liveblog/syndication/syndication.py [hunk 6]
```diff
@@ -52,18 +54,24 @@ def _cursor(self, resource=None):
     def _get_blog(self, blog_id):
         return self._cursor('blogs').find_one({'_id': ObjectId(blog_id)})
 
-    def is_syndicated(self, consumer_id, producer_blog_id, consumer_blog_id):
+    def _lookup(self, consumer_id, producer_blog_id, consumer_blog_id):
```
Comment (djbrd-sourcefabric) [inline server/liveblog/syndication/syndication.py]: If this is only used once, I wouldn't put it in a separate method

File: server/liveblog/syndication/syndication.py [hunk 7]
```diff
@@ -143,25 +159,35 @@ class SyndicationOut(Resource):
...
+        'default': None
     }
 }
 
 
 class SyndicationInService(BaseService):
     notification_key = 'syndication_in'
 
-    def is_syndicated(self, producer_id, producer_blog_id, consumer_blog_id):
+    def _lookup(self, producer_id, producer_blog_id, consumer_blog_id):
```
Comment (djbrd-sourcefabric) [inline server/liveblog/syndication/syndication.py]: Again, no need for a separate method here I think

File: server/liveblog/syndication/producer.py (hunk 2 above)
Comment (mscam) [inline server/liveblog/syndication/producer.py]: blog_token should be returned to the client only when necessary, for security reasons, only by using SSL secure connection.

File: server/liveblog/syndication/producer.py (hunk 4 above)
Comment (mscam) [inline server/liveblog/syndication/producer.py]: auto_publish can be updated by sending PATCH request to the blueprint url. This is not actually implemented by the javascript client, but it's available.

File: server/liveblog/syndication/syndication.py (hunk 5 above)
Comment (mscam) [inline server/liveblog/syndication/syndication.py]: Unfortunately python-eve doesn't set the default value I have selected (None) and set by default to datetime.datetime.now(). I am investigating about this issue.

**Pull request merged**